core/services/fluxmonitor/flux_monitor.go

```diff
@@ -668,10 +657,16 @@ func (p *PollingDeviationChecker) respondToAnswerUpdatedLog(log *contracts.LogAn
 // need to poll and submit an answer to the contract regardless of the deviation.
 //
 // Only invoked by the CSP consumer on the single goroutine for thread safety.
-func (p *PollingDeviationChecker) respondToNewRoundLog(log *contracts.LogNewRound) {
-	// The idleThreshold resets when a new round starts
+func (p *PollingDeviationChecker) respondToNewRoundLog(log contracts.LogNewRound) {
+	// The idleTimer resets when a new round starts
 	if !p.idleThreshold.IsInstant() {
-		p.idleTicker = time.After(p.idleThreshold.Duration())
+		p.idleTimer = MakeIdleTimer(log, p.idleThreshold, utils.Clock{})
+		select {
+		case <-p.idleTimer:
+			// TODO: How to handle immediately expired timer here?
```

Comment:
@RyanRHall this still needs to be dealt with

Reply:
Sounds good to me

Reply:
I also think that's the best solution :+1: